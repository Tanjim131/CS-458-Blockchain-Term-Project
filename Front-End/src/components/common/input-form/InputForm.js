import React, { FC } from "react";
import { Form, Input } from "antd";
import { InfoCircleOutlined } from "@ant-design/icons";

import "./InputForm.less";

/**
 * This is used for InputForm
 *
 * @usage: <InputForm formItemName={string} formItemLabel={string} rulesMsg={string} required={boolean} tooltipTitle={string} placeholderText={string} />
 */
const InputForm = (props) => {
  const {
    className,
    formItemName,
    formItemLabel,
    rulesMsg = "Fields required",
    required = true,
    tooltipTitle,
    placeholderText,
    clearField = false,
    allowedRegExp = "",
    dependencies,
    initialValue,
    type,
    autocomplete
  } = props;

  return (
    <>
      <Form.Item
        name={formItemName}
        label={formItemLabel}
        className={className ? className : "add-document-form-field"}
        initialValue={initialValue}
        rules={[
          {
            required,
            pattern: new RegExp(allowedRegExp, "i"),
            message: rulesMsg,
          },
          ({ getFieldValue }) => ({
            validator(_, value) {
              if (dependencies) {
                if (!value || getFieldValue("accountNumber") === value) {
                  return Promise.resolve();
                }
                return Promise.reject(
                  "The two account number that you entered do not match!",
                );
              }
              return Promise.resolve();

            },
          }),
        ]}
        tooltip={
          tooltipTitle
            ? {
              title: tooltipTitle,
              icon: <InfoCircleOutlined />,
              placement: "rightTop",
              overlayClassName: "tooltip-add-document",
            }
            : ""
        }
        dependencies={dependencies}
      >
        <Input allowClear={clearField} placeholder={placeholderText} type={type} autoComplete={autocomplete}/>
      </Form.Item>
    </>
  );
};

export default InputForm;
