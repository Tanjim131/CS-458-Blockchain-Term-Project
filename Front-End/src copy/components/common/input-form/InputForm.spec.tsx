import React from 'react';
import { shallow } from "enzyme";
import InputForm from "./InputForm";
import { shallowToJson } from "enzyme-to-json";

describe('InputForm Component', () => {
    let wrapper: any;

    beforeAll(() => {
        wrapper = shallow(
            <InputForm
                formItemName="formItemName"
                formItemLabel="formItemLabel"
                rulesMsg="ruleMsg"
                required={true}
                tooltipTitle="title"
                placeholderText="placeHolderText"
                clearField={true}
                allowedRegExp="regex"
                className="className"
            />
        )
    })

    it('Check if component exists or not and capture snapshot', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    })

    it("should check when the className is empty", () =>{
        wrapper.setProps({ rulesMsg  : "Fields required", required: true, clearField: false, allowedRegExp:"", className:""});
        expect(wrapper).toBeTruthy()
    })

})