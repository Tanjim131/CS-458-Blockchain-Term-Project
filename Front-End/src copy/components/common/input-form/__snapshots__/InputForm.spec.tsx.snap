// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputForm Component Check if component exists or not and capture snapshot 1`] = `
<Fragment>
  <FormItem
    className="className"
    label="formItemLabel"
    name="formItemName"
    rules={
      Array [
        Object {
          "message": "ruleMsg",
          "pattern": /regex/,
          "required": true,
        },
        [Function],
      ]
    }
    tooltip={
      Object {
        "icon": <ForwardRef(InfoCircleOutlined) />,
        "overlayClassName": "tooltip-add-document",
        "placement": "rightTop",
        "title": "title",
      }
    }
  >
    <Input
      allowClear={true}
      placeholder="placeHolderText"
      type="text"
    />
  </FormItem>
</Fragment>
`;
